src/mon/OSDMonitor.cc

```diff
@@ -4586,6 +4693,90 @@ void OSDMonitor::tick()
   if (do_propose ||
       !pending_inc.new_pg_temp.empty())  // also propose if we adjusted pg_temp
     propose_pending();
+
+  if (ceph_using_tcmalloc() && pcm != nullptr) {
+    _adjust_cache_ratios();
+    pcm->balance();
+    pcm->tune_memory();
+    _set_new_cache_sizes();
+  }
+}
+
+void OSDMonitor::_set_new_cache_sizes()
+{
+  uint64_t cache_size = 0;
+  int64_t inc_alloc = 0;
+  int64_t full_alloc = 0;
+  int64_t kv_alloc = 0;
+
+  if (pcm != nullptr && rocksdb_binned_kv_cache != nullptr) {
+    cache_size = pcm->get_tuned_mem();
+    inc_alloc = inc_cache->get_committed_size();
+    full_alloc = full_cache->get_committed_size();
+    kv_alloc = rocksdb_binned_kv_cache->get_committed_size();
+  }
+
+  uint64_t max_inc_cache = static_cast<uint64_t>(
+    inc_alloc / mon_memory_target / inc_cache->get_avg_bytes_per_osdmap());
+  // set full cache size to be equal to inc cache size
```

Comment:
why make them equal when the ratio settings exist? perhaps we can get rid of the ratio settings for full/incremental

Reply:
On the OSD side we allow the ratios to be tweaked both so that user can adjust them when autotuning is disabled and also so that the autotuner will use those ratios as sort of "fairshare" targets when caches compete at given priority levels.  If full/incremental will always be given equal priority you could get rid of the ratios.  I suspect we may want to leave them as dev options (and then trust the ratios), but I'll defer to @jdurgin as he knows the mon better than I do.

Reply:
Ratio settings are removed.